identity-account-storage/src/stronghold/wrapper.rs

```diff
@@ -0,0 +1,143 @@
+
+use std::path::Path;
+
+use identity_iota_core::did::IotaDID;
+use iota_stronghold::Client;
+use iota_stronghold::ClientError;
+use iota_stronghold::KeyProvider;
+use iota_stronghold::SnapshotPath;
+use iota_stronghold::Stronghold as IotaStronghold;
+use tokio::sync::RwLock;
+use zeroize::Zeroize;
+
+use crate::stronghold::error::ClientOperation;
+use crate::stronghold::error::SnapshotOperation;
+use crate::stronghold::error::StrongholdError;
+use crate::stronghold::ClientPath;
+use crate::stronghold::StrongholdResult;
+use crate::utils::derive_encryption_key;
+use crate::utils::EncryptionKey;
+use crate::Result;
+
+/// The implementation of the `Storage` interface using `Stronghold`.
+///
+/// Stronghold is a secure storage for sensitive data. Secrets that are stored inside a Stronghold
+/// can never be read, but only be accessed via cryptographic procedures. Data written into a Stronghold
+/// is persisted in snapshots which are encrypted using the provided password.
+#[derive(Debug)]
+pub struct Stronghold {
+  pub(crate) stronghold: IotaStronghold,
+  snapshot_path: SnapshotPath,
+  key_provider: KeyProvider,
+  pub(crate) index_lock: RwLock<()>,
+  pub(crate) dropsave: bool,
+}
+
+impl Stronghold {
+  /// Constructs a Stronghold storage instance.
+  ///
+  /// Arguments:
+  ///
+  /// * `path`: path to a local Stronghold snapshot file. Will be created if it does not exist.
+  /// * `password`: password for the Stronghold snapshot file. If this is cloned from a reference,
+  /// zeroization of that reference is strongly recommended.
+  /// * `dropsave`: persist all changes when the instance is dropped. Default: true.
+  pub async fn new<T>(path: &T, mut password: String, dropsave: Option<bool>) -> Result<Self>
+  where
+    T: AsRef<Path> + ?Sized,
+  {
+    let stronghold: IotaStronghold = IotaStronghold::default();
+
+    let mut key: EncryptionKey = derive_encryption_key(&password);
+    password.zeroize();
+
+    let key_provider = KeyProvider::try_from(key.to_vec()).map_err(StrongholdError::Memory)?;
+    key.zeroize();
+
+    // If the snapshot file exists, we load it.
+    // If it doesn't we write data into the in memory `Stronghold` and only persist to disk on first write.
+    let snapshot_path: SnapshotPath = if path.as_ref().exists() {
+      let snapshot_path = SnapshotPath::from_path(path);
+
+      stronghold
+        .load_snapshot(&key_provider, &snapshot_path)
+        .map_err(|err| StrongholdError::Snapshot(SnapshotOperation::Read, snapshot_path.clone(), err))?;
+
+      snapshot_path
+    } else {
+      SnapshotPath::from_path(path)
+    };
+
+    Ok(Self {
+      stronghold,
+      snapshot_path,
+      key_provider,
+      index_lock: RwLock::new(()),
+      dropsave: dropsave.unwrap_or(true),
+    })
+  }
+
+  /// Returns whether dropsave is enabled.
+  pub fn dropsave(&self) -> bool {
+    self.dropsave
+  }
+
+  /// Sets whether dropsave is enabled.
+  pub fn set_dropsave(&mut self, dropsave: bool) {
```

Comment:
ultra-nit: 
```suggestion
  pub fn dropsave_mut(&mut self, dropsave: bool) {
```

Reply:
Mh, this is just a setter, and does not return a mutable reference to `dropsave`, so the naming matches Rust conventions. `clippy` would probably complain about `dropsave_mut`, too.